src/Starcounter.Database.Extensions/PreCommitTransactor.cs

```diff
@@ -35,6 +35,11 @@ public PreCommitTransactor(ITransactor transactor, IOptions<PreCommitOptions> pr
         protected override IDatabaseContext EnterContext(IDatabaseContext db) => new PreCommitContext(db);
 
         protected override void LeaveContext(IDatabaseContext db, bool exceptionThrown)
-            => ((PreCommitContext)db).ExecutePreCommitHooks(hookOptions);
+        {
+            if (!exceptionThrown && db is PreCommitContext context)
```

Comment:
Previously it didn't check if `db` is `PreCommitContext`. It would be a bug if it's not, wouldn't it?

Reply:
Haha, that's quite funny. Right after I changed to that, from the straight cast, I came to think that "Hm, this is probably not going to pass a review from Vadim, it change in semantics from actually failing to ignoring", and right I was. :)

Yes, I also actually think the previous approach is better. Although the one with `is` look nicer. But let's change it back. 👍